guardrails/datatypes.py

```
@@ -344,6 +344,8 @@ def validate(self, key: str, value: Any, schema: Dict) -> Dict:
 
         # Validate the selected choice
         selected_key = value
+        if selected_key not in schema:
```

Ideally, this should raise a validation error, since the expected key doesn't exist in the generated dictionary. I'm curious, did this come up during your experimentation?

Something like an `EventDetail` with something like `f'{selected_key}' not in schema.`

---

Good catch, this is #123 which was accidentally included (a while ago I put it on my main branch and built guardrails from there).